in_toto/in_toto_sign.py

```diff
@@ -0,0 +1,326 @@
+                                 "signature in file")
+
+  sign_parser.add_argument("-i", "--infix", action="store_true",
+                            help="Infix keyID in the filename while "
+                                 "dumping, when -i the file will be dumped "
+                                 "as original.<keyID>.link/layout, "
+                                 "else original.link/layout")
+
+  sign_parser.add_argument("-v", "--verbose", dest="verbose",
+                            help="Verbose execution.", default=False,
+                            action="store_true")
+
+  sign_parser.add_argument("-d", "--destination", type=str, help="Filepath "
+                           "to which the output file should be dumped")
+
+  sign_parser.add_argument("-k", "--keys", nargs="+", type=str, required=True,
+                            help="Path to the key file/files")
+
+  verify_parser = subparsers.add_parser('verify', help='Verify the '
+                                        'link/layout file.')
+
+  verify_parser.add_argument("signablepath", type=str,
+                            help="path to the signable file")
+
+  verify_parser.add_argument("-v", "--verbose", dest="verbose",
+                            help="Verbose execution.", default=False,
+                            action="store_true")
+
+  verify_parser.add_argument("-k", "--keys", nargs="+", type=str,
+                            required=True, help="Path to the key file/files")
+
+
+  args = parser.parse_args()
+  return args
+
+def main():
+  """
+  First calls parse_args to parse the arguments, and then calls either
+  add_sign or add_replace depending upon the arguments. Based on the
+  arguments it then dumps the corresponding file.
+  """
+  args = parse_args()
+  length = len(args.keys)
+  rsa_key = in_toto.util.import_rsa_key_from_file(args.keys[length-1])
+  source_file_name = os.path.basename(args.signablepath)
+
+  if args.verbose:
+    log.logging.getLogger().setLevel(log.logging.INFO)
+
+  if args.subparser_name == 'sign':
+    try:
+      if args.destination and args.infix:
+        log.error('Please specify one and only one out of infix and '
+                  'destination!')
+        sys.exit(1)
+      else:
+        if args.replace_sig:
+          signable_object = replace_sign(args.signablepath, args.keys)
+        else:
+          signable_object = add_sign(args.signablepath, args.keys)
+
+        if signable_object._type == 'layout':
+          if not args.destination:
+            log.warn('Layout file, ignoring --infix while dumping...')
+            log.info('Dumping in the current working directory...')
+            signable_object.dump(source_file_name)
+            sys.exit(0)
+          else:
+            file_name = os.path.join(args.destination)
+            signable_object.dump(file_name)
+            log.info('Dumping at ' + str(file_name) + '...')
+            sys.exit(0)
+
+        if signable_object._type == 'Link':
```

Comment:
I guess we could even take it a step further and write the last three lines only once for layouts and links. :)
e.g.:
```python
log.info("Dumping {0} to '{1}'' ...".format(signable_object._type.lower(), path))
signable_object.dump(path)
sys.exit(0)
```